core/modules/node/node.block.inc

```diff
@@ -68,8 +71,9 @@ class NodeBlock extends Block {
     }
 
     // Add 'submitted by' settings.
-    if (!empty($this->settings['display_submitted'])) {
+    if ($this->settings['display_submitted']) {
       $content['#display_submitted'] = TRUE;
```

Comment:
What should happen here is `#display_submitted` should be passed to `template_preprocess_node()` as `$variables['display_submitted']`, and then from there to `node.tpl.php` as `$display_submitted`. We shouldn't need to (and don't want to) modify the `$node` object to include display information.